src/main/java/taiga/util/pbChange/PBHelper.java

```diff
@@ -0,0 +1,97 @@
+package taiga.util.pbChange;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import settings.Settings;
+import taiga.api.EpicsAPI;
+import taiga.api.HistoryAPI;
+import taiga.api.SprintAPI;
+import taiga.api.UserStoryAPI;
+import taiga.model.query.epic.EpicDetail;
+import taiga.model.query.history.History;
+import taiga.model.query.sprint.Sprint;
+import taiga.model.query.sprint.UserStoryDetail;
+
+public class PBHelper {
+
+    private List<Sprint> sprintList;
+    private List<UserStoryDetail> userStoryList;
+    private List<EpicDetail> epicList;
+    private List<History> userStoryHistoryList = new ArrayList<>();
+
+    UserStoryAPI userStoryAPI = new UserStoryAPI();
+    SprintAPI sprintAPI = new SprintAPI();
+    EpicsAPI epicAPI = new EpicsAPI();
+    HistoryAPI historyAPI = new HistoryAPI();
+    
+    private int projectId;
+
+    public PBHelper() {
```

Comment:
This class could be completely removed, just utilizing the calls directly in `PBChangeCalculator` since there is nothing special to PB Change that requires its own helper class. Also, some api calls should be substituted with cheaper things (e.g., replace `fetchSprintList` with `Settings.get().getAppModel().getCurrentProject().get().getSprints()`)

Reply:
Removed the fetchSprintList as it is duplicate code, leaving the calss as shoving API calls into the calculate class is overreaching and seems like a violation of SRP

Reply:
I don't think it will be an SRP violation. As things stand, these helper methods still need to be called alongside the methods in `PBChangeCalculator` to get the full filtered list of PB change items. This would delegate the work to one of two places:

- the UI would be the one to construct the calculations, which would be a violation of SRP (the data should be ready by the time the UI requests it) and difficult to test if need be 
- Another helper class, adds a layer of complexity. The problem with this is that if a fundamental part of how we decide to do the calculation changes, the extra helper class would likely change also, which indicates the responsibility is still shared among all involved classes.

Reply:
Good points, refactored needed methods into PBChangeCalculator and removed PBHelper class.